src/include/sof/drivers/ipc.h

```diff
@@ -52,12 +52,12 @@ struct ipc_msg;
 
 /* validates internal non tail structures within IPC command structure */
 #define IPC_IS_SIZE_INVALID(object)					\
-	object.hdr.size == sizeof(object) ? 0 : 1
+	(object).hdr.size == sizeof(object) ? 0 : 1
 
 /* convenience error trace for mismatched internal structures */
 #define IPC_SIZE_ERROR_TRACE(class, object)				\
 	trace_error(class, "ipc: size %d expected %d",			\
-		    object.hdr.size, sizeof(object))
+		    (object).hdr.size, sizeof(object))
 
```

Comment:
Why do we need this? Even if it is legit please add it to a separate patch with proper explanation.

Reply:
I assume that `(object)->hdr` where `object` is a macro parameter never required explanation and is even recommended. Not sure why reference would require explanations vs. dereference which does not.

Reply:
@mmaka1 i see, so I think this deserves a separate patch with this explanation you gave here in the commit message.

Reply:
Done